File: app/src/main/java/com/dewonderstruck/apps/ashx0/ui/product/filtering/Grouping.java
package com.dewonderstruck.apps.ashx0.ui.product.filtering;

import com.dewonderstruck.apps.ashx0.utils.Constants;
import com.dewonderstruck.apps.ashx0.viewobject.Category;
import com.dewonderstruck.apps.ashx0.viewobject.SubCategory;

import java.util.ArrayList;
import java.util.LinkedHashMap;
import java.util.List;

public class Grouping {

    public String catId,subCatId;
    public LinkedHashMap<Category,List<SubCategory>> map = new LinkedHashMap<>();

    public LinkedHashMap<Category,List<SubCategory>> group(List<Category> categoryList,List<SubCategory> subCategoryList)
    {
        map.clear();


        for (int i = 0;i < categoryList.size();i++) {

            List<SubCategory> subCategories = new ArrayList<>();

            SubCategory subCategory = new SubCategory(Constants.ZERO,"",Constants.CATEGORY_ALL,"","","","","","","" +
                    "","",null,null);
            subCategories.add(0,subCategory);

            catId = categoryList.get(i).id;

            for (int j = 0;j < subCategoryList.size();j++)
            {
                subCatId = subCategoryList.get(j).catId;

                if (catId.equals(subCatId))
                {
                    subCategories.add(subCategoryList.get(j));
                }
            }

            if(!subCategories.isEmpty())
            {

                map.put(categoryList.get(i),subCategories);
            }
        }

        return map;
    }
}

File: app/src/main/java/com/dewonderstruck/apps/ashx0/ui/product/filtering/Grouping.kt
package com.dewonderstruck.apps.ashx0.ui.product.filtering

import com.dewonderstruck.apps.ashx0.utils.Constants
import com.dewonderstruck.apps.ashx0.viewobject.Category
import com.dewonderstruck.apps.ashx0.viewobject.SubCategory
import java.util.*

class Grouping {
    var catId: String? = null
    var subCatId: String? = null
    var map = LinkedHashMap<Category, List<SubCategory>>()
    fun group(categoryList: List<Category>, subCategoryList: List<SubCategory>): LinkedHashMap<Category, List<SubCategory>> {
        map.clear()
        for (i in categoryList.indices) {
            val subCategories: MutableList<SubCategory> = ArrayList()
            val subCategory = SubCategory(Constants.ZERO, "", Constants.CATEGORY_ALL, "", "", "", "", "", "", "" +
                    "", "", null, null)
            subCategories.add(0, subCategory)
            catId = categoryList[i].id
            for (j in subCategoryList.indices) {
                subCatId = subCategoryList[j].catId
                if (catId == subCatId) {
                    subCategories.add(subCategoryList[j])
                }
            }
            if (!subCategories.isEmpty()) {
                map[categoryList[i]] = subCategories
            }
        }
        return map
    }
}